core/src/main/java/org/elasticsearch/index/query/MatchQueryParser.java

```diff
@@ -69,8 +64,18 @@ public Query parse(QueryShardContext context) throws IOException, QueryParsingEx
 
         Object value = null;
         float boost = AbstractQueryBuilder.DEFAULT_BOOST;
-        MatchQuery matchQuery = new MatchQuery(context);
         String minimumShouldMatch = null;
+        String analyzer = null;
```

Comment:
I think for this one you really need null, I have the same in the query_string query too.